botorch/utils/probability/lin_ess.py

```diff
@@ -316,130 +331,84 @@ def _get_cart_coords(self, nu: Tensor, theta: Tensor) -> Tensor:
         """
         return self._z * torch.cos(theta) + nu * torch.sin(theta)
 
-    def _find_rotated_intersections(self, nu: Tensor) -> Tuple[Tensor, Tensor]:
-        r"""Finds rotated intersections.
+    def _trim_intervals(self, left, right):
```

Comment:
Could you add type annotations here and throughout?

Reply:
Sure, will do.